src/openms/include/OpenMS/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.h

```diff
@@ -0,0 +1,270 @@
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_TRANSFORMATIONS_FEATUREFINDER_FEATUREFINDERIDENTIFICATIONALGORITHM_H
+#define OPENMS_TRANSFORMATIONS_FEATUREFINDER_FEATUREFINDERIDENTIFICATIONALGORITHM_H
+
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/ANALYSIS/SVM/SimpleSVM.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/ChromatogramExtractor.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/ElutionModelFitter.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPickedHelperStructs.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/EGHTraceFitter.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/GaussTraceFitter.h>
+#include <OpenMS/ANALYSIS/TARGETED/TargetedExperiment.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/TransformationDescription.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/DATAACCESS/SimpleOpenMSSpectraAccessFactory.h>
+
+#include <vector>
+#include <fstream>
+#include <map>
+
+namespace OpenMS
+{
+
+class OPENMS_DLLAPI FeatureFinderIdentificationAlgorithm :
+  public DefaultParamHandler
+{
+public:
+  /// default constructor
+  FeatureFinderIdentificationAlgorithm(); 
+
+  /// Main method for actual FeatureFinder
+  /// External ids may be empty and greatly speed up the detectin process
+  void run(
+    std::vector<PeptideIdentification> peptides,
+    std::vector<ProteinIdentification> proteins,
+    std::vector<PeptideIdentification> peptides_ext,
+    std::vector<ProteinIdentification> proteins_ext,
+    FeatureMap & features
```

Comment:
No space before "&".

Reply:
Actually, it is not my editor (vim) that does that but me ;)
That way, I find it much easier to spot if references are passed.

I have seen "Type const" being used more often recently and I think newer 
compiler sometimes complain about using: "const auto &" instead of "auto const &".

I don't care too much about these two style issues and would in principle be willing to adopt whatever you like. But I would like to make these style questions more of a community decision.
Most important thing is to be consistent than to decide for A or B,

Reply:
We had the discussion about spaces around "*" and "&" before. It is still in our Coding Conventions (https://github.com/OpenMS/OpenMS/wiki/Coding-conventions), although not all the examples on that page are consistent with it: 
> If function arguments are pointers or references, the pointer or reference qualifier is appended to the variable type. It should not prefix the variable name.

The problem with spaces on both sides is that it makes it look like a mathematical operator (more of an issue for "*").

The vast majority of the code in OpenMS uses "const Type", so absent a good reason to the contrary, we should stick to that - as you said, consistency is most important.